File: Submissions/228_Summary_Ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> result;
        if(nums.empty()) return result;
        
        int begin = nums[0];
        int end = begin;
        for(int i=1; i<nums.size(); ++i){
            if (nums[i] == nums[i-1] + 1){
                end = nums[i];
            } else {
                addRange(begin, end, &result);
                
                begin = nums[i];
                end = begin;
            }
        }
        addRange(begin, end, &result);
        
        return result;
    }
    
    void addRange(int begin, int end, vector<string>* result){
        if(begin == end){
            result->push_back(to_string(end));
        } else {
            string str = to_string(begin) + "->" + to_string(end);
            result->push_back(str);
        }
    }
};

File: Kotlin/228_Summary_Ranges.kt
class Solution {
    fun summaryRanges(nums: IntArray): List<String> {
        if(nums.isEmpty()) return emptyList()
        
		val result = arrayListOf<String>()
        var begin = nums[0]
		var end = nums[0]
		val builder = StringBuilder()

        for(i in 1 until nums.size) {
            if(nums[i] == nums[i-1] + 1) {
                end = nums[i]
            } else {
                result.add(getRange(begin, end))
                
                begin = nums[i]
                end = begin
            }
            
        }
        result.add(getRange(begin, end))
        
		return result
    }

    fun getRange(begin: Int, end: Int): String{
        return if (begin == end) "$begin" else "$begin->$end"
    }
}